spec/models/memberships/join_zusatzsektion_spec.rb

```diff
@@ -0,0 +1,240 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2024, Schweizer Alpen-Club. This file is part of
+#  hitobito_sac_cas and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito_sac_cas
+
+require 'spec_helper'
+
+describe Memberships::JoinZusatzsektion do
+  def create_role(key, role, owner: person, **attrs)
+    group = key.is_a?(Group) ? key : groups(key)
+    role_type = group.class.const_get(role)
+    Fabricate(role_type.sti_name, group: group, person: owner, **attrs)
+  end
+
+  it 'initialization fails if no subgroup exists' do
+    expect do
+      described_class.new(Group::Sektion.new, :person, :date)
+    end.to raise_error('missing subgroup')
+
+    expect do
+      described_class.new(Group::Ortsgruppe.new, :person, :date)
+    end.to raise_error('missing subgroup')
+
+    expect do
+      described_class.new(groups(:bluemlisalp), :person, :date)
+    end.not_to raise_error
+
+    expect do
+      described_class.new(groups(:bluemlisalp_ortsgruppe_ausserberg), :person, :date)
+    end.not_to raise_error
+  end
+
+  describe 'validations' do
+    let(:person) { Fabricate(:person) }
+    let(:join_section) { groups(:bluemlisalp) }
+    let(:errors) { obj.errors.full_messages }
+    let(:date) { Date.new(2024, 6, 14) }
+
+    subject(:obj) { described_class.new(join_section, person, date) }
+
+    it 'is invalid if person is not an sac member' do
+      expect(obj).not_to be_valid
+      expect(errors).to eq ['Person muss Sac Mitglied sein']
+    end
+
+    it 'is valid with membership in different section' do
+      create_role(:matterhorn_mitglieder, 'Mitglied')
+      expect(obj).to be_valid
+    end
+
+    it 'is invalid and contains all validation and role validation errors' do
+      allow(obj).to receive(:prepare_roles) do |person|
+        # invalid role without group
+        Fabricate.build(Group::SektionsMitglieder::Mitglied.sti_name,
+                        person: person)
+      end
+
+      expect(obj).not_to be_valid
+      expect(errors).to eq ['Person muss Sac Mitglied sein',
+                            "#{person}: Group muss ausgefüllt werden"]
```

Comment:
das wird ja schon in der base class getestet, braucht es das hier auch nochmals?

Reply:
imho schadet das nicht, gibt uns sicherheit, dass base class verhalten erhalten geblieben ist